solidity-v1/dashboard/src/sagas/keep-to-t-staking.js

```diff
@@ -0,0 +1,209 @@
+    yield put(
+      showModal({
+        modalType: MODAL_TYPES.ThresholdAuthorizationLoadingModal,
+        modalProps: {
+          text: "Please, authorize in your wallet",
+        },
+      })
+    )
+
+    try {
+      yield call(sendTransaction, {
+        payload: {
+          contract: stakingContract,
+          methodName: "authorizeOperatorContract",
+          args: [operator, operatorContractAddress],
+          options: {
+            onTransactionHashAction: (txHash) =>
+              showModal({
+                modalType: MODAL_TYPES.ThresholdAuthorizationLoadingModal,
+                modalProps: {
+                  txHash,
+                },
+              }),
+          },
+        },
+      })
+    } catch (err) {
+      yield put(hideModal())
+      return
+    }
+
+    yield put(thresholdContractAuthorized(operator))
+  }
+
+  yield put(
+    showModal({
+      modalType: MODAL_TYPES.ThresholdStakeConfirmationLoadingModal,
+      modalProps: {
+        text: "Please, confirm in your wallet",
+      },
+    })
+  )
+
+  try {
+    yield call(sendTransaction, {
+      payload: {
+        contract: Keep.keepToTStaking.thresholdStakingContract.instance,
+        methodName: "stakeKeep",
+        args: [operator],
+        options: {
+          onTransactionHashAction: (txHash) =>
+            showModal({
+              modalType: MODAL_TYPES.ThresholdStakeConfirmationLoadingModal,
+              modalProps: {
+                txHash,
+              },
+            }),
+        },
+      },
+    })
+  } catch (err) {
+    yield put(
+      showModal({
+        modalType: MODAL_TYPES.AuthorizedButNotStakedToTWarningModal,
+      })
+    )
+    return
+  }
+
+  yield put(stakedToT(operator))
+}
+
+export function* watchAuthorizeAndStakeKeepToT() {
+  yield takeEvery(STAKE_KEEP_TO_T, authorizeAndStakeKeepToT)
+}
+
+function* fetchThresholdAuthData(action) {
+  try {
+    const {
+      payload: { address },
+    } = action
+    yield put({ type: FETCH_THRESHOLD_AUTH_DATA_START })
+    const data = yield call(
+      thresholdAuthorizationService.fetchThresholdAuthorizationData,
+      address
+    )
+    yield put({
+      type: FETCH_THRESHOLD_AUTH_DATA_SUCCESS,
+      payload: data,
+    })
+  } catch (error) {
+    yield* logErrorAndThrow(FETCH_THRESHOLD_AUTH_DATA_FAILURE, error)
+  }
+}
+
+export function* watchFetchThresholdAuthData() {
+  yield takeOnlyOnce(
+    FETCH_THRESHOLD_AUTH_DATA_REQUEST,
+    identifyTaskByAddress,
+    fetchThresholdAuthData
+  )
+}
+
+export function* watchFetchThresholdAuthDataSuccess() {
+  yield takeEvery(
+    FETCH_THRESHOLD_AUTH_DATA_SUCCESS,
+    processThresholdAuthorizationNotification
+  )
+}
+
+function* processThresholdAuthorizationNotification(action) {
+  const stakesAvailableToStakeOnThreshold = action.payload.filter(
+    (stake) => !stake.isStakedToT
+  ).length
+
+  const inflectionOfAStakeWord =
+    stakesAvailableToStakeOnThreshold === 1 ? "stake" : "stakes"
```

Comment:
I believe the `StakeReadyToBeStakedToT` message component should handle this. So we can pass just `stakesAvailableToStakeOnThreshold` prop to the message component and render the correct title on the component side.